From 5bbaf33b2500b9f1ca1137663924a722e00aad57 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Wed, 29 Jun 2016 12:08:31 -0400
Subject: [PATCH] rtl8xxxu: Add trxff_boundary for 8188e

The 8188e presumably has a 10K buffer, but leave space for TX report
or WOL pattern.

Signed-off-by: Andrea Merello <andrea.merello@gmail.com>
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
@@ -225,6 +225,11 @@ struct rtl8xxxu_fileops rtl8188eu_fops =
 	.power_on = rtl8188eu_power_on,
 	.reset_8051 = rtl8xxxu_reset_8051,
 	.usb_quirks = rtl8188e_usb_quirks,
+	/*
+	 * Use 9K for 8188e normal chip
+	 * Max RX buffer = 10K - max(TxReportSize(64*8), WOLPattern(16*24))
+	 */
+	.trxff_boundary = 0x23ff,
 	.total_page_num = TX_TOTAL_PAGE_NUM_8188E,
 	.page_num_hi = TX_PAGE_NUM_HI_PQ_8188E,
 	.page_num_lo = TX_PAGE_NUM_LO_PQ_8188E,
